addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1624,6 +1624,51 @@ options.registry.anchor = options.Class.extend({
     },
 });
 
+/**
+ * Controls box properties.
+ */
+options.registry.Box = options.Class.extend({
+    /**
+     * @override
+     */
+    selectClass(previewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (!widgetValue && params.possibleValues.includes('shadow')) {
+            this.$target.css('box-shadow', '');
+        }
+    },
+
+    /**
+     * @see this.selectClass for parameters
+     */
+    setShadow(previewMode, widgetValue, params) {
+        const hasClass = this.$target.hasClass(params.shadowClass);
+        const inset = widgetValue === 'inset' ? widgetValue : '';
+        // if the class doesn't change, the customization is conserved.
+        let values = hasClass ? this.$target.css('box-shadow').replace('inset', '') + ` ${inset}` : undefined;
+        this.$target.toggleClass(params.shadowClass, !!widgetValue);
+        // if we are adding the class, we use the class style.
+        values = !hasClass && widgetValue ? this.$target.css('box-shadow').replace('inset', '') + ` ${inset}` : values;
```

Comment:
Why a ternary form here? Use a simple if...

Actually, I don't understand why `this.$target.css('box-shadow').replace('inset', '') + ' ${inset}'` is called twice, should not it be something simple like this?
```javascript
this.$target.toggleClass(...);
if (widgetValue) {
    const boxShadow = this.$target.css('box-shadow').replace(...);
    ...
}
```
why would this lose the custo ?